lib/Sema/CSSimplify.cpp

```diff
@@ -1926,6 +1926,10 @@ ConstraintSystem::matchExistentialTypes(Type type1, Type type2,
             return getTypeMatchFailure(locator);
         }
```

Comment:
@theblixguy I think we can do this: 
``` 
else { // no elements in the path
   auto *anchor = locator.getAnchor();
   if (!(anchor && isa<AssignExpr>(anchor)))
     return getTypeMatchFailure(locator);
}
```

Since `MissingContextualConformanceFailure::diagnoseAsError` handles `AssignExpr` with empty path.

Reply:
Okay, I was thinking to do 

```c++
if (getConstraintLocator(locator)->getPath().empty())
  return getTypeMatchFailure(locator);
```

but that works as well.

Reply:
We shouldn’t need a transform for that, should be identical :)

Reply:
Also just want to mention, we should probably get locators like that fixed to point to call expression instead, so we don’t have to locate parent to figure out the context...

Reply:
Yeah, it does point to one of the branches in the if/ternary statement, but I think the path should include a call/argument element as well?

Reply:
Ideally it should be something like `<call expr> -> function argument #0 -> if expr -> {condition, then, else}` to incapsulate all information.